nixos/modules/services/networking/openvpn.nix

```diff
@@ -161,6 +166,27 @@ in
             '';
           };
 
+          authUserPass = mkOption {
```

Comment:
There should be a warning, that using this option, will put user and password world-readable in nix store.
To avoid this a user would have to specify `auth-user-pass` in `services.openvpn.servers.<name>.config` instead.

Reply:
Setting the credentials inside the NixOS configuration would really help in many situations, even with the compromise of a world-readable username/password. 

I'm not entirely sure though if we should provide an additional option to configure a custom credentials-file, as this is just a generic OpenVPN option. I guess it would make more sense when, at some point, the OpenVPN configuration can be done with the Nix language entirely. The only reason for this change is, that it is not possible to set the credentials from the NixOS configuration otherwise. What do you think?

Reply:
Also, I guess that this use-case is not the only one which could benefit from credentials being stored somewhat safely inside the Nix store. Is there really no option to accomplish that / something planned?

Reply:
See https://github.com/NixOS/nix/issues/8 and https://github.com/NixOS/rfcs/pull/5
It's not trivial to solve.